pkgs/build-support/php/build-php.nix

```diff
@@ -0,0 +1,93 @@
+{ stdenvNoCC, lib, writeTextDir, composer, php }@inputs:
+
+{ pname
+, version
+, buildInputs ? [ ]
+, nativeBuildInputs ? [ ]
+, src
+, vendorOutputHash
+, passthru ? { }
+, php ? inputs.php
+, composer ? inputs.composer
+, ...
+}@args:
+
+stdenvNoCC.mkDerivation (
+  args // (let
+  inherit version src;
+
+  vendor = stdenvNoCC.mkDerivation {
+    inherit src version;
+
+    pname = "${pname}-vendor";
+
+    # See https://github.com/NixOS/nix/issues/6660
+    dontPatchShebangs = true;
+
+    nativeBuildInputs = [
+      composer
+    ];
+
+    strictDeps = true;
+
+    buildPhase = let
+      # See https://getcomposer.org/doc/03-cli.md#composer-home-config-json
+      composerHome = let
+        content = {
+          config = {
+            autoloader-suffix = "${builtins.replaceStrings ["-"] [""] pname}";
+          };
+        };
+      in
+        writeTextDir "/config.json" (builtins.toJSON content);
+    in
+      ''
+      runHook preBuild
+
+      set -x
+
+      COMPOSER_HOME=${composerHome} \
+      COMPOSER_VENDOR_DIR=vendor \
```

Comment:
This is the default value https://getcomposer.org/doc/03-cli.md#composer-vendor-dir

Reply:
I'm doing that to override any potential local values from `composer.json`.